codec/decode.go

```diff
@@ -734,6 +734,15 @@ func (f *decFnInfo) kStruct(rv reflect.Value) {
 					} else {
 						d.decodeValue(si.field(rv, true), nil)
 					}
+				} else if fti.ufh {
```

Comment:
I implemented the non-length-prefixed map case, but added a panic for the non-map case.